tpl/tplimpl/template_ast_transformers.go

```diff
@@ -112,7 +112,7 @@ func getParseTree(templ tpl.Template) *parse.Tree {
 }
 
 const (
-	partialReturnWrapperTempl = `{{ $_hugo_dot := $ }}{{ $ := .Arg }}{{ with .Arg }}{{ $_hugo_dot.Set ("PLACEHOLDER") }}{{ end }}`
+	partialReturnWrapperTempl = `{{ $_hugo_dot := $ }}{{ $ := .Arg }}{{ range (slice .Arg) }}{{ $_hugo_dot.Set ("PLACEHOLDER") }}{{ end }}`
```

Comment:
I suspect it would be easier to understand if you make `contextWrapper.Arg` a `[]interface{}` -- and just add a comment there explaining why it's a slice with a single value.

Reply:
I gave this a go, though my attempt at a comment depended on explaining what we were doing with the `partialReturnWrapperTempl`. In the end, I found it simpler to add a comment above `partialReturnWrapperTempl` while leaving `contextWrapper.Arg` as it was. How does that sound?